recipes/GKE Setup/README.md

```diff
@@ -0,0 +1,105 @@
+```shell
+gcloud container clusters get-credentials "<cluster-id>"
+```
+Before running the `deploy.sh` script, you must
+[create a BigQuery dataset](https://cloud.google.com/bigquery/docs/datasets#create-dataset) (or use an existing dataset)
+and a [Google Storage bucket](https://cloud.google.com/storage/docs/creating-buckets).
+
+As a default, the script expects the name of the dataset to be `fullstory-export-data` and the
+name of the bucket to be `$GOOGLE_PROJECT-hauser`, but feel free to change these.
+
+If you did not use the default dataset and bucket names, or want to customize the name of the service account,
+feel free to edit the following variables at the top of the `deploy.sh` script.
+```shell
+BIGQUERY_DATASET="fullstory-export-data"
+SERVICE_NAME="hauser-svc"
+GCS_BUCKET="$GOOGLE_PROJECT-hauser"
+HAUSER_VERSION="latest"
+```
+To find your FullStory Org Id, see [help docs](https://help.fullstory.com/hc/en-us/articles/360047075853-How-do-I-find-my-FullStory-Org-Id-).
+You will also need to create or get a [FullStory API Key](https://help.fullstory.com/hc/en-us/articles/360020624834-Where-can-I-find-my-API-key-).
+
+The `HAUSER_VERSION` variable corresponds to the docker image tag. See [docker hub](https://hub.docker.com/r/fullstorydev/hauser) for available tags.
+
+## The deploy script
+The `deploy.sh` script in this directory uses the gcloud SDK and associated CLI's to
+create a service account, add the permissions, and create a Kubernetes deployment that
+runs hauser.
+
+Assuming you have completed the prerequisites, you can run the deploy script with:
+```shell
+./deploy.sh -p "<my_project_id>" -o "<my_org_id>" -k "<fs_api_key"
+```
+
+Note: The script is idempotent, so if it fails, you can simply run it again after resolving any issues.
+
+If a service account does not exist, the script will confirm before creating it. Assuming the service account creation
+succeeded, the kubernetes included `hauser.yaml` will be rendered with the provided variables and diffed against your
+current GKE cluster. The script will ask for confirmation before applying the diff.
+
+If successful, you can check that the hauser pod is running with:
+```shell
+kubectl get pods -lapp=hauser -c hauser
+```
+which should return with something like
+```
+NAME                     READY   STATUS    RESTARTS   AGE
+hauser-<pod_id_suffix>   2/2     Running   0          1m
+```
+
+To verify that everything is working as expected, you can also check the logs with
+```shell
+kubect logs -lapp=hauser -c hauser
+```
+
+## Service accounts
+This setup is based on using a service account for the hauser deployment.
+In short, this is recommended to adhere to the principal of east privilege.
+To read more about service accounts in Google, see [Google's documentation](https://cloud.google.com/iam/docs/service-accounts)
+
+The `create-svc-account.sh` script will create/validate that the service account
```

Comment:
It doesn't look like users are intended to run this directly, but instead just let `deploy.sh` run it. Is that right? Maybe merge into `deploy.sh` if that is the case? Or document it a little more (required inputs/params) and/or make it more usable to be run independently?

Reply:
Yeah, I think I might just merge it. That'll definitely simplify the validation bits.